nomad/leader.go

```diff
@@ -774,16 +774,47 @@ func (s *Server) restorePeriodicDispatcher() error {
 			continue
 		}
 
+		// We skip if the job doesn't allow overlap and there are already
+		// instances running
+		needed, err := s.isNewEvalNeeded(job)
+		if err != nil {
+			return fmt.Errorf("failed to get job status: %v", err)
+		}
+
+		if !needed {
+			continue
+		}
+
 		if _, err := s.periodicDispatcher.ForceRun(job.Namespace, job.ID); err != nil {
 			logger.Error("force run of periodic job failed", "job", job.NamespacedID(), "error", err)
 			return fmt.Errorf("force run of periodic job %q failed: %v", job.NamespacedID(), err)
 		}
-		logger.Debug("periodic job force runned during leadership establishment", "job", job.NamespacedID())
+
+		logger.Debug("periodic job force run during leadership establishment", "job", job.NamespacedID())
 	}
 
 	return nil
 }
 
+// isNewEvalNeeded checks if the job allows for overlap and if there are already
+// instances of the job running in order to determine if a new evaluation needs to
+// be created upon periodic dispatcher restore
+func (s *Server) isNewEvalNeeded(job *structs.Job) (bool, error) {
```

Comment:
Not to bikeshed too much, but this function is specific to periodic jobs, therefore it might be nice to have a more descriptive name. Otherwise, this seems like a generic job helper function at first glance.